packages/interaction/src/InteractionManager.js

```diff
@@ -1438,34 +1453,57 @@ export default class InteractionManager extends EventEmitter
                 }
             }
         }
+        else if (isTouch)
+        {
+            const isDown = trackingData && trackingData.touchDown;
+
+            if (hit)
+            {
+                this.dispatchEvent(displayObject, 'touchend', interactionEvent);
+
+                if (isDown)
+                {
+                    this.dispatchEvent(displayObject, 'tap', interactionEvent);
+                    // because we can confirm that the touch happened on this object, flag for later emit of pointertap
+                    isTap = true;
+                }
+            }
+            else
+            {
+                this.dispatchEvent(displayObject, 'touchendoutside', interactionEvent);
+            }
+
+            if (isDown)
+            {
+                trackingData.touchDown = false;
+            }
+        }
 
         // Pointers and Touches, and Mouse
         if (hit)
         {
             this.dispatchEvent(displayObject, 'pointerup', interactionEvent);
-            if (isTouch) this.dispatchEvent(displayObject, 'touchend', interactionEvent);
 
-            if (trackingData)
+            // emit pointertap if the mouse/touch block decided it was a tap
+            if (isTap)
             {
-                // emit pointertap if not a mouse, or if the mouse block decided it was a tap
-                if (!isMouse || isMouseTap)
```

Comment:
I don't think this old code was the cleanest way to go about it, but this `!isMouse` -> emit pointertap did mean that a pointer that wasn't a mouse/pen or a touch did emit pointertap appropriately. It looks like the new code won't do that in the off chance that a pointer device that is none of those three is used. Added another `else` after the `if (isMouse) {} else if (isTouch) {}` that just checks that case and sets `isTap` appropriately should solve it.